src/main/java/org/takes/misc/Href.java

```diff
@@ -64,7 +64,7 @@
     public Href(final String txt) {
         this.params = new ConcurrentHashMap<String, List<String>>(0);
         final URI link = URI.create(txt);
-        final String query = link.getQuery();
+        final String query = link.getRawQuery();
```

Comment:
@dmzaytsev let's add also note in method comment how this method handles encoded queries